user_sync/connector/directory_ldap.py

```diff
@@ -377,7 +438,33 @@ def format_ldap_query_string(query, **kwargs):
             escaped_args[k] = six.text_type('').join(escaped_list)
         return query.format(**escaped_args)
 
+    def format_group_user_filter (self, group_dn):
+        """
+         :type group_dn: str
+         :rtype str
+         """
+        group_member_subfilter = self.format_ldap_query_string(self.options['group_member_filter_format'], group_dn=group_dn)
+        if not group_member_subfilter.startswith('('):
+            group_member_subfilter = six.text_type('(') + group_member_subfilter + six.text_type(')')
+        user_subfilter = self.options['all_users_filter']
+        if not user_subfilter.startswith('('):
+            user_subfilter = six.text_type('(') + user_subfilter + six.text_type(')')
+        group_user_filter = six.text_type('(&') + group_member_subfilter + user_subfilter + six.text_type(')')
+        return group_user_filter
 
+    @staticmethod
+    def is_dn_within_base_dn_scope(base_dn, dn):
```

Comment:
We should consider using a proper DN parser instead of trying to do it ourselves.  `python-ldap` should have one, but I don't know how easy it will be to use.  If it's reasonable to do so, I would love to see that used here before we merge the PR.  If it's not feasible, I'm OK waiting until the ldap3 switch because ldap3 [also has a DN parser](https://ldap3.readthedocs.io/ldap3.utils.dn.html).